providers/openfeature-meta_provider/spec/openfeature/meta_provider_spec.rb

```diff
@@ -8,19 +8,31 @@
       let(:flag_key) { "first_match_#{type}" }
 
       it "returns from first" do
+        expected_result = OpenFeature::SDK::Provider::ResolutionDetails.new(
+          value: first_matched_value,
+          reason: OpenFeature::SDK::Provider::Reason::STATIC,
+          flag_metadata: {"provider_name" => "In-memory Provider"}
```

Comment:
I'm assuming the `provider_name` represents the underlying provider that "won". If that's the case, is there a name that would better represent that behavior?

Reply:
I could also see calling it `matched_provider` (I'll update with that to see how that feels). One follow-on thought was to consider having this metadata set by the provider all the time in the SDK instead of specifically for the meta provider. That's a larger change/not in the spec, so I'm going to keep it here for now.